pkgs/by-name/ec/echidna/package.nix

```
@@ -0,0 +1,115 @@
+{
+  lib,
+  stdenv,
+  haskellPackages,
+  fetchFromGitHub,
+  # build dependencies
+  slither-analyzer,
+}:
+
+haskellPackages.mkDerivation rec {
+  pname = "echidna";
+  version = "2.2.6";
+
+  src = fetchFromGitHub {
+    owner = "crytic";
+    repo = "echidna";
+    tag = "v${version}";
+    sha256 = "sha256-5nzis7MXOqs0bhx2jrEexjZYZI2qY6D0D7AWO+SPs+A=";
+  };
+
+  isExecutable = true;
+
+  libraryToolDepends = with haskellPackages; [
```

Not tied to either the library or the executable component.

```suggestion
  buildTools = with haskellPackages; [
```

---

done.